packages/lab/src/__tests__/__e2e__/date-picker/DatePicker.single.cy.tsx

```diff
@@ -48,44 +48,38 @@ describe("GIVEN a DatePicker where selectionVariant is single", () => {
   };
 
   it("SHOULD support validation", () => {
-    const selectedDateChangeSpy = cy.stub().as("selectedDateChangeSpy");
+    const selectionChangeSpy = cy.stub().as("selectionChangeSpy");
     cy.mount(
       <SingleWithFormField
         selectionVariant={"single"}
-        onSelectedDateChange={selectedDateChangeSpy}
+        onSelectionChange={selectionChangeSpy}
         locale={testLocale}
       />,
     );
     // Simulate entering a valid date
     cy.findByRole("textbox").click().clear().type(initialDateValue);
     cy.realPress("Tab");
     cy.findByRole("textbox").should("have.value", initialDateValue);
-    cy.get("@selectedDateChangeSpy").should("have.been.calledOnce");
-    cy.get("@selectedDateChangeSpy").should(
-      "have.been.calledWith",
-      initialDate,
-    );
+    cy.get("@selectionChangeSpy").should("have.been.calledOnce");
+    cy.get("@selectionChangeSpy").should("have.been.calledWith", initialDate);
     // Simulate entering an invalid date
     cy.findByRole("textbox").click().clear().type("bad date");
     cy.realPress("Tab");
-    cy.get("@selectedDateChangeSpy").should("have.been.calledTwice");
-    cy.get("@selectedDateChangeSpy").should("have.been.calledWith", null);
+    cy.get("@selectionChangeSpy").should("have.been.calledTwice");
+    cy.get("@selectionChangeSpy").should("have.been.calledWith", null);
```

Comment:
When I debug the test here, `clear()` and `type` is not reflected in `DateInputSingle`'s `apply` function, i.e., `parse()` is called with "05 Jan 2025" rather than "bad date"